### PR TITLE
[15.0][IMP] delivery_package_number: Allow set package numbers at confirm and new labels

### PR DESCRIPTION
**Pull request by CarlosRoca13**
cc @Tecnativa TT45428

ping @sergio-teruel @chienandalu

### CODE REVIEW

File: delivery_package_number/models/stock_picking_type.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Copyright 2023 Tecnativa - Carlos Roca
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import fields, models
+
+
+class StockPickingType(models.Model):
+    _inherit = "stock.picking.type"
+
+    force_set_number_of_packages = fields.Boolean()
```
Comment (chienandalu) [inline delivery_package_number/models/stock_picking_type.py:9]: It'd be nice to be able to choose the report

**Review by chienandalu — APPROVED** [review]

**Review by chienandalu — COMMENTED** [review]
Sorry, it was comment, not approve :sweat_smile:

**Comment by CarlosRoca13** [comment]
Added the option :smile:

File: delivery_package_number/wizard/stock_number_package_validate_wiz.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        if self.print_package_label:
+            report = (
+                self.pick_ids.picking_type_id.report_number_of_packages
+                or self.env.ref(
+                    "delivery_package_number.action_delivery_package_number_report"
+                )
+            )
+            report_action = report.report_action(self.pick_ids)
+            report_action.update({"close_on_report_download": True})
+            return report_action
```
Comment (chienandalu) [inline delivery_package_number/wizard/stock_number_package_validate_wiz.py]: Shouldn't it be attached to the picking. What do you think @sergio-teruel ?

Comment (sergio-teruel) [reply delivery_package_number/wizard/stock_number_package_validate_wiz.py]: This labels are generated by Odoo directly so I think tha t is not needed store into db...

**Review by chienandalu — APPROVED** [review]
:+1:

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge major

**Comment by pedrobaeza** [comment]
Please fw-port it to v16.

**Pull request merged**